marshmallow_dataclass/__init__.py

```diff
@@ -88,14 +91,16 @@ def dataclass(_cls: type = None, *,
     Point(x=0.0, y=0.0)
     """
     dc = dataclasses.dataclass(_cls, repr=repr, eq=eq, order=order, unsafe_hash=unsafe_hash, frozen=frozen)
-    return add_schema(dc) if _cls else lambda cls: add_schema(dc(cls))
+    return add_schema(dc, custom_base_schema_class) if _cls else lambda cls: add_schema(dc(cls), custom_base_schema_class)
 
 
-def add_schema(clazz: type) -> type:
+def add_schema(_cls = None, custom_base_schema_class: Optional[Type[marshmallow.Schema]] = None):
```

Comment:
Please do not remove type hints. I agree that just `(type) -> type` was not great, and you can improve it by using generics, but please at least leave the existing annotation.

Reply:
`custom_base_schema_class`-> `base_schema`